From 9d38a220cc47ea6f84e606d7e9cfca6cefc6b8b7 Mon Sep 17 00:00:00 2001
From: Waiman Long <Waiman.Long@hp.com>
Date: Wed, 14 May 2014 14:15:14 -0400
Subject: [PATCH 13/17] vfs: Remove second variable named error in __dentry_path
References: FATE#317271
Git-commit: a8323da0366d3398eda62741d2ac1130c8a172ed
Patch-mainline: v3.14-rc1

In commit  232d2d60aa5469bb097f55728f65146bd49c1d25
Author: Waiman Long <Waiman.Long@hp.com>
Date:   Mon Sep 9 12:18:13 2013 -0400

    dcache: Translating dentry into pathname without taking rename_lock

The __dentry_path locking was changed and the variable error was
intended to be moved outside of the loop.  Unfortunately the inner
declaration of error was not removed. Resulting in a version of
__dentry_path that will never return an error.

Remove the problematic inner declaration of error and allow
__dentry_path to return errors once again.

This patch is a backport of the upstream commit:

  a8323da0366d3398eda62741d2ac1130c8a172ed

Cc: stable@vger.kernel.org
Cc: Waiman Long <Waiman.Long@hp.com>
Signed-off-by: "Eric W. Biederman" <ebiederm@xmission.com>
Signed-off-by: Al Viro <viro@zeniv.linux.org.uk>
Acked-by: Jeff Mahoney <jeffm@suse.com>
---
 fs/dcache.c |    1 -
 1 files changed, 0 insertions(+), 1 deletions(-)

diff --git a/fs/dcache.c b/fs/dcache.c
index 80e6427..865eced 100644
--- a/fs/dcache.c
+++ b/fs/dcache.c
@@ -2900,7 +2900,6 @@ restart:
 	read_seqbegin_or_lock(&rename_lock, &seq);
 	while (!IS_ROOT(dentry)) {
 		struct dentry *parent = dentry->d_parent;
-		int error;
 
 		prefetch(parent);
 		error = prepend_name(&end, &len, &dentry->d_name);
-- 
1.7.1

